core/sector_storage/impl/manager_impl.cpp

```diff
@@ -778,6 +785,208 @@ namespace fc::sector_storage {
+      remoteGetSector(response, matches[1], matches[2]);
+    } else {
+      logger_->error("Unknown route: {}", target);
+      response.result(http::status::internal_server_error);
+    }
+  }
+
+  void ManagerImpl::createResponseDelete(
+      const http::request<http::dynamic_body> &request,
+      http::response<http::dynamic_body> &response) {
+    std::regex delete_sector_rgx(R"(\/remote\/([\w]+)\/([\w-]+))");
+    std::smatch matches;
+
+    auto target = request.target().to_string();
+    if (std::regex_search(
+            target.cbegin(), target.cend(), matches, delete_sector_rgx)) {
+      response.set(http::field::content_type, "text/plain");
+      remoteRemoveSector(response, matches[1], matches[2]);
+    } else {
+      logger_->error("Unknown route: {}", target);
+      response.result(http::status::internal_server_error);
+    }
+  }
+
+  void ManagerImpl::remoteStatFs(http::response<http::dynamic_body> &response,
+                                 const fc::primitives::StorageID &storage_id) {
+    auto maybe_stat = local_store_->getFsStat(storage_id);
+    if (maybe_stat.has_error()) {
+      if (maybe_stat
+          == fc::outcome::failure(
+              fc::sector_storage::stores::StoreErrors::kNotFoundStorage)) {
+        response.result(404);
+        return;
+      }
+      logger_->error("Error remote stat fs: {}", maybe_stat.error().message());
+      response.result(500);
+      return;
+    }
+
+    auto json_doc = fc::api::encode(maybe_stat.value());
+
+    auto maybe_json = fc::codec::json::format(&json_doc);
+
+    if (maybe_json.has_error()) {
+      logger_->error("Error with encoding in remote stat fs: {}",
+                     maybe_json.error().message());
+      response.result(500);
+      return;
+    }
+
+    response.set(http::field::content_type, "application/json");
+    beast::ostream(response.body())
+        .write(fc::common::span::bytestr(maybe_json.value().data()),
+               maybe_json.value().size());
+  }
+
+  void ManagerImpl::remoteGetSector(
+      http::response<http::dynamic_body> &response,
+      const std::string &type,
+      const std::string &sector) {
+    auto maybe_sector = fc::primitives::sector_file::parseSectorName(sector);
+
+    if (maybe_sector.has_error()) {
+      logger_->error("Error remote get sector: {}",
+                     maybe_sector.error().message());
+      response.result(500);
+      return;
+    }
+
+    auto maybe_type = fc::primitives::sector_file::fromString(type);
+    if (maybe_type.has_error()) {
+      logger_->error("Error remote get sector: {}",
+                     maybe_type.error().message());
+      response.result(500);
+      return;
+    }
+
+    // Proof type is 0 because we don't allocate anything
+    auto maybe_paths = local_store_->acquireSector(
+        maybe_sector.value(),
+        static_cast<RegisteredProof>(0),
+        fc::primitives::sector_file::SectorFileType::FTNone,
+        maybe_type.value(),
```

Comment:
```suggestion
        maybe_type.value(),
        fc::primitives::sector_file::SectorFileType::FTNone,
```
Please check `existing` and `allocate` order

Reply:
Fixed